### PR TITLE
Add GTFS Fetch

### PR DESCRIPTION
**Pull request by alanna-zhou**
Issue #9 Add gtfs data fetch to a /gtfs endpoint to replace this data fetch in node in the future

### CODE REVIEW

File: gtfs.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        column_names = row
+        continue
+      route = {}
+      for index, column in enumerate(column_names):
+        route[column] = row[index]
+      gtfs_data.append(route)
+
+def needs_fetch():
+  cmd = "wget -N " + GTFS_URL
+  p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
```
Comment (meganle) [inline gtfs.py]: try to avoid single letter vars

File: gtfs.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import csv
+from datetime import datetime
+from subprocess import PIPE, Popen, STDOUT
+import threading
+import zipfile
+
+GTFS_URL = 'https://s3.amazonaws.com/tcat-gtfs/tcat-ny-us.zip'
+TCAT_NY_US = 'tcat-ny-us'
+TEN_SECS = 10
```
[the remark this answers is not in the document]
Comment (kq-li) [reply gtfs.py]: name constant after its use, not after what it is

File: gtfs.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+gtfs_data = None
+
+def fetch_gtfs(event):
+  try:
+    if needs_fetch():
+      unzip_gtfs()
+      extract_gtfs()
+  except:
```
[the remark this answers is not in the document]
Comment (kq-li) [reply gtfs.py]: dO nOt usE bArE ExCePt

File: gtfs.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    print(traceback.format_exc())
+  threading.Timer(TEN_SECS, fetch_gtfs, [event]).start()
+
+def extract_gtfs():
+  global gtfs_data
+  gtfs_data = []
+  with open('{}/routes.txt'.format(TCAT_NY_US)) as csv_file:
```
Comment (kq-li) [inline gtfs.py]: https://www.python.org/dev/peps/pep-0498/

File: gtfs.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    for index, row in enumerate(csv_reader):
+      if index == 0:
+        column_names = row
+        continue
+      route = {}
+      for index, column in enumerate(column_names):
+        route[column] = row[index]
+      gtfs_data.append(route)
+
+def needs_fetch():
```
[the remark this answers is not in the document]
Comment (kq-li) [reply gtfs.py]: someone clearly needs fetch

File: gtfs.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  cmd = "wget -N " + GTFS_URL
+  p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
+  output = p.stdout.read().decode('utf-8')
+  return not 'Omitting download' in output
+
+def unzip_gtfs():
+  zip_ref = zipfile.ZipFile('{}.zip'.format(TCAT_NY_US), 'r')
```
Comment (kq-li) [inline gtfs.py]: see above for f strings

**Comment by young-k** [comment]
Doesn't the Docker container also need to install `wget`? Can you verify whether or not this works by testing it with Docker locally?

```
docker build -t [some image name] .
docker run -p "5000:5000" -it [some image name]
```

File: gtfs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alanna-zhou) [reply gtfs.py]: i made that comment on young's pr before about generally refactoring constants but he told me to still do this??

File: gtfs.py (hunk 4 above)
Comment (alanna-zhou) [reply gtfs.py]: but i thought young always wanted strings to be formatted this way

Comment (young-k) [reply gtfs.py]: looks like times are a'changin

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+version: "3.4"
+
+services: 
+  live-tracking:
+   build: .
+   environment:
+     - TOKEN=ZG9QcmQ1SXE2M3RmSUVfSzI3QzJyOThTSVhNYTpIY2g3bXh1dUh3Y29HdE5WTmxXNTBxNDlGdThh
+   ports:
+     - "5000:5000"
+
```
Comment (young-k) [inline docker-compose.yml]: remove newline

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+version: "3.4"
+
+services: 
+  live-tracking:
+   build: .
+   environment:
+     - TOKEN=ZG9QcmQ1SXE2M3RmSUVfSzI3QzJyOThTSVhNYTpIY2g3bXh1dUh3Y29HdE5WTmxXNTBxNDlGdThh
```
Comment (young-k) [inline docker-compose.yml]: delete this

File: gtfs.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    unzip_gtfs()
+    extract_gtfs()
+  threading.Timer(TEN_SECONDS, fetch_gtfs, [event]).start()
+
+def extract_gtfs():
+  global gtfs_data
+  with open(f'{TCAT_NY_US}/routes.txt') as csv_file:
+    csv_reader = csv.reader(csv_file, delimiter=',')
+    column_names = next(csv_reader)
+    for index, row in enumerate(csv_reader):
```
Comment (young-k) [inline gtfs.py]: we're not using `index` here, why not just `for `loop

**Pull request merged**